### PR TITLE
Moved unit testing to Docker

### PR DESCRIPTION
**Pull request by TomHAnderson**
This is the logical next step from https://github.com/doctrine/DoctrineMongoODMModule/pull/197

Provide a Docker environment for development use and run all unit tests through the same Docker environment.

### CODE REVIEW

File: .docker/ascii.art [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+|.       . ..ooooo..     .|
+|    :oooooooooooooo.     |
+|    .:::::::oooooo:      |
+|.         .:oooo:.      .|
+|o.       :oooo:.       .o|
+|oo.      :o::.        .oo|
+|ooo:..              .oooo|
+|oooooo:..       ..:oooooo|
++-------------------------+
+ Doctrine Mongo ODM Module
```
Comment (alcaeus) [inline .docker/ascii.art]: While the module may still be called "DoctrineMongoODMModule", we should start using "MongoDB" as the proper name for the database.

Comment (malarzm) [reply .docker/ascii.art]: Lucky you, it'll still fit ;)

**Comment by webimpress** [comment]
Now I agree here with @Ocramius and his comment from #197. I can see the point that it could be useful on local, but on travis it is much easier to use services. Especially when we support just PHP 7.1+ travis configuration is quite easy (see #206).

**Comment by greg0ire** [comment]
Consider removing the dot in `.docker`, it only makes things harder to grep.

File: .docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+RUN apk add --no-cache \
+	autoconf \
+	make \
+	g++ \
+	bash \
+	git \
+	openssl-dev
+
+RUN echo -e 'memory_limit=2G' > /usr/local/etc/php/conf.d/memory.ini
+RUN set -o pipefail && curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
```
Comment (greg0ire) [inline .docker/Dockerfile]: I think the recommended way to do this nowadays is to copy the binary from the official Docker image, see https://hub.docker.com/_/composer#php-extensions . If it slows the build down to much though, consider installing it that way: https://getcomposer.org/doc/faqs/how-to-install-composer-programmatically.md#how-do-i-install-composer-programmatically-

**Comment by TomHAnderson** [comment]
Merged via PR https://github.com/doctrine/DoctrineMongoODMModule/pull/212

**Pull request closed without merging**